sstable.go

```diff
@@ -32,9 +36,35 @@ func NewSSTable(file *os.File) *SSTable {
 		return nil
 	}
 	sst.index = index
+
+	sst.minKey = sst.getFirstKeyFromFile()
+	sst.maxKey = index.entries[len(index.entries)-1].key
+
 	return sst
 }
 
+func (s *SSTable) getFirstKeyFromFile() []byte {
+	firstIndexEntry := s.index.entries[0]
+	length := binary.LittleEndian.Uint32(firstIndexEntry.value[4:8])
+
+	buf, err := s.readBlockAt(0, length)
```

Comment:
### Runtime Errors
Handle potential runtime errors when reading blocks from the file. Specifically, ensure that `s.readBlockAt(0, length)` is checked for errors and handle them appropriately. Consider logging the error and returning a more informative error message:
```go
buf, err := s.readBlockAt(0, length)
if err != nil {
    log.Printf("Error reading block: %v", err)
    return nil
}
```